### ISSUE TITLE
Trace: Bad address balance during SUICIDE

### ISSUE THREAD

**Issue by iFA88** [ledgerwatch/erigon#3953]
Hey everybody!

I found a trace bug with the client version `2022.04.2-beta` on linux.
The `0x9e0f7c64a5bf1fc9f3d7b7963cf23f74e3d2c0b2b3f35f26df031954e5581179` transaction creates a contract on address `0xe76d0a8810bfc2f43a30894ac1e4185b4bc761f4` with 0 wei, after that will be call'd with 1 wei which does a SUICIDE op code and the trace says that the address has 2 wei balance: https://gist.github.com/iFA88/4b76023d9f91215cc088adb6defb7bfd#file-gistfile1-txt-L74
I expected just 1 wei. OpenEthereum trace gives correct 1 wei.

Network:
`Ethereum Ropsten testnet`

Request:
```
curl --data '{"method":"trace_transaction","params":["0x9e0f7c64a5bf1fc9f3d7b7963cf23f74e3d2c0b2b3f35f26df031954e5581179"],"id":1,"jsonrpc":"2.0"}' -H "Content-Type: application/json" localhost:8545
```

Thanks for the support!

**Comment by iFA88** [comment]
Issue is still available with v2022.05.02 release.

**Comment by AskAlexSharov** [comment]
thank you. we will take a look on it later.

**Comment by iFA88** [comment]
FYI the issue is still available with v2022.05.04 release.

**Comment by iFA88** [comment]
Sadly the issue is still available with `v2022.06.02` release.
